ZenPacks/zenoss/Microsoft/Windows/datasources/ClusterDataSource.py

```diff
@@ -0,0 +1,296 @@
+
+        psClusterCommands.append(
+            "$diskInfo = Get-Disk | Get-Partition | Select DiskNumber, PartitionNumber,"
+            "@{{Name='Volume';Expression={{Get-Volume -Partition $_ | Select -ExpandProperty ObjectId}};}},"
+            "@{{Name='DriveLetter';Expression={{Get-Volume -Partition $_ | Select -ExpandProperty DriveLetter}};}},"
+            "@{{Name='FileSystemLabel';Expression={{Get-Volume -Partition $_ | Select -ExpandProperty FileSystemLabel}};}},"
+            "@{{Name='Size';Expression={{Get-Volume -Partition $_ | Select -ExpandProperty Size}};}},"
+            "@{{Name='SizeRemaining';Expression={{Get-Volume -Partition $_ | Select -ExpandProperty SizeRemaining}};}};"
+            "$clsDisk = get-clusterresource -errorvariable diskerr -erroraction"
+            " 'silentlycontinue'| where {{ $_.ResourceType -eq 'Physical Disk'}};"
+            "if( -Not $diskerr){{"
+            "foreach ($disk in $clsDisk) {{"
+            "$founddisk = $diskInfo | where {{ $_.FileSystemLabel -eq $disk.Name}};"
+            "if ($founddisk -ne $null) {{"
+            "$diskowner = $disk.OwnerNode.Name;"
+            "$disknumber = $founddisk.DiskNumber;"
+            "$diskpartition = $founddisk.PartitionNumber;"
+            "$disksize = $founddisk.Size;"
+            "$disksizeremain = $founddisk.SizeRemaining;"
+            "$diskvolume = $founddisk.Volume.substring(3);"
+            "$physicaldisk = New-Object -TypeName PSObject -Property @{{"
+            "Id = $diskvolume.substring(8, $diskvolume.length-10);"
+            "Name = $disk.Name;VolumePath = $diskvolume;"
+            "OwnerNode = $diskowner;DiskNumber = $disknumber;"
+            "PartitionNumber = $diskpartition;Size = $disksize;"
+            "FreeSpace = $disksizeremain;State = $disk.State;}};"
+            "$physicaldisk | foreach {{{}}} }};}};}}".format(cluster_id_items)
+        )
+
+        script = "\"& {{{}}}\"".format(''.join(psClusterCommands))
+        return pscommand, script
+
+    @defer.inlineCallbacks
+    def collect(self, config):
+        conn_info = createConnectionInfo(config.datasources[0])
+        command = SingleCommandClient(conn_info)
+
+        pscommand, script = self.build_command_line()
+        results = yield command.run_command(pscommand, script)
+
+        defer.returnValue(results)
+
+    @save
+    def onSuccess(self, results, config):
+        log.debug('Cluster collection results: {}'.format(results))
+        data = self.new_data()
+        for result in results.stdout:
+            # ignore any empty lines
+            if len(result) <= 0:
+                continue
+            try:
+                comp, state = result.split('|')
+            except Exception:
+                log.debug('Unable to parse cluster result {} on {}'.format(result, config.id))
+                continue
+            comp = prepId(comp)
+            data['values'][comp]['state'] = cluster_state_value(state), int(time.mktime(time.localtime()))
```

Comment:
I think you can just pass 'N' and avoid any issues with time zones on that mktime/localtime thing (i'm not sure, but that might assume that the local timezone is GMT to give you the right value)